cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1036,7 +1038,7 @@ private KafkaVersionChange getKafkaVersionChange(StatefulSet kafkaSts) {
 
                         return kafkaSetOperations.maybeRollingUpdate(newSts, pod -> {
                             log.info("{}: Upgrade: Maybe patch + rolling update of {}: Pod {}", reconciliation, stsName, pod.getMetadata().getName());
-                            return true;
+                            return "Upgrade: Maybe patch + rolling update of " + name + ": Pod " + pod.getMetadata().getName();
```

Comment:
There's no maybe at this point. The function is returning a non-null reason to roll, so the pod will be rolled. The interesting information is that the rolling is due to upgrade. Also worth including that this is phase 1 of a possibly 2 roll process.